### PR TITLE
Fix PHP Warning on Checking Constants

### PR DESCRIPTION
**Pull request by hoseinrafiei**
Fixes https://github.com/eventespresso/eea-wpuser-integration/issues/45 issue.

### CODE REVIEW

File: EED_WP_Users_SPCO.module.php [hunk 1]
```diff
@@ -118,7 +118,7 @@ public static function set_hooks()
     public static function set_hooks_admin()
     {
         // hook into filters/actions done on ajax but ONLY EE_FRONT_AJAX requests
-        if (EE_FRONT_AJAX) {
+        if (defined('EE_FRONT_AJAX') && EE_FRONT_AJAX) {
```
Comment (tn3rb) [inline EED_WP_Users_SPCO.module.php:121]: yup 👍🏻 

just an FYI, if the `EventEspresso\core\services\request\Request` class is loaded then you can also use it to check for AJAX coming from the EE frontend like so:

```php
if ($this->request->isFrontAjax()) {
```

**Review by tn3rb — APPROVED** [review]
Looks Great thnx 💯

**Comment by hoseinrafiei** [comment]
@tn3rb 

I wasn't able to merge this one due to the errors of auto check. The "Squash and Merge" button is disabled.
Could you please help me on this? If you guide me how to fix the errors, It will be better since I can do it myself.

I also added `yoast/phpunit-polyfills` package as a dev dependency to the composer file but it didn't help. Maybe I missed something.

**Review by tn3rb — APPROVED** [review]

**Comment by hoseinrafiei** [comment]
@tn3rb I cannot merge it. Maybe I don't have permission to do that.

Not Able to Merge

**Comment by tn3rb** [comment]
done. thnx @hoseinrafiei 

unit tests foradd-ons got messed up after the release of WP 5.9 and I haven't had a chance yet to solve that issue, but this was such an obvious straightforward fix that i'm ok with merging despite tests not running.

**Comment by Pebblo** [comment]
@tn3rb is this not going to cause issues in later versions of PHP? 

If PHP8 throws a fatal because it no longer assumes which `EE_FRONT_AJAX` we meant then even if we check for the definition first, doesn't that just mean this code won't run in PHP8+?

**Comment by tn3rb** [comment]
no the error had nothing to do with PHP 8
not sure why it suddenly started getting upset about that but that code should never have been the way it was.
Hossein's fix is how the code should have been written from the start.

**Comment by Pebblo** [comment]
The warning we get currently in PHP7 and below doesn't, but the fatal which PHP8 throws:

```
[04-Mar-2022 20:22:08 UTC] PHP Fatal error:  Uncaught Error: Undefined constant "EE_FRONT_AJAX" in /home/pebbloco/public_html/ee4/wp-content/plugins/eea-wp-user-integration/EED_WP_Users_SPCO.module.php:121
Stack trace:
#0 /home/pebbloco/public_html/ee4/wp-content/plugins/event-espresso-core-reg/core/EE_Config.core.php(1236): EED_WP_Users_SPCO::set_hooks_admin()
#1 /home/pebbloco/public_html/ee4/wp-content/plugins/event-espresso-core-reg/core/EE_Config.core.php(1009): EE_Config->_initialize_modules()
#2 /home/pebbloco/public_html/ee4/wp-includes/class-wp-hook.php(307): EE_Config->initialize_shortcodes_and_modules()
#3 /home/pebbloco/public_html/ee4/wp-includes/class-wp-hook.php(331): WP_Hook->apply_filters()
#4 /home/pebbloco/public_html/ee4/wp-includes/plugin.php(474): WP_Hook->do_action()
#5 /home/pebbloco/public_html/ee4/wp-content/plugins/event-espresso-core-reg/core/EE_System.core.php(1224): do_action()
#6 /home/pebbloco/public_html/ee4/wp-includes/class-wp-hook.php(307): EE_System->core_loaded_and_ready()
#7 /home/pebbloco/public_html/ee4/wp-includes/class-wp-hook.php(331): WP_Hook->apply_filters()
#8 /home/pebbloco/public_html/ee4/wp-includes/plugin.php(474): WP_Hook->do_action()
#9 /home/pebbloco/public_html/ee4/wp-settings.php(587): do_action()
#10 /home/pebbloco/public_html/ee4/wp-config.php(98): require_once('/home/pebbloco/...')
#11 /home/pebbloco/public_html/ee4/wp-load.php(50): require_once('/home/pebbloco/...')
#12 /home/pebbloco/public_html/ee4/wp-admin/admin-ajax.php(22): require_once('/home/pebbloco/...')
#13 {main}
  thrown in /home/pebbloco/public_html/ee4/wp-content/plugins/eea-wp-user-integration/EED_WP_Users_SPCO.module.php on line 121
```

Suggests that with PHP8 the code will be skipped right? The error:

`Uncaught Error: Undefined constant "EE_FRONT_AJAX"`

Now we are doing:  `if (defined('EE_FRONT_AJAX') && EE_FRONT_AJAX) {`

But it's not defined in PHP8, so this will just be skipped, right?

(I've bust my local PHP8 install to play around with it right now, fix that shortly but posting this up)

**Pull request merged**